src/worker/main.ts

```diff
@@ -23,15 +23,15 @@ const client = new Redis(getRedisInfo("client").redisOptions);
 const subscriber = new Redis(getRedisInfo("subscriber").redisOptions);
 const logger = getLogger("worker.main");
 
-function measureElapsedTime(startTime, tags) {
-	const endTime = Date.now();
-	const timeDiff = endTime - startTime;
-	statsd.histogram(metricHttpRequest().jobDuration, timeDiff, tags);
+function measureElapsedTime(job: Queue.Job, tags) {
+	statsd.histogram(metricHttpRequest().jobDuration, job.finishedOn - job.processedOn, tags);
 }
 
 const queueOpts: QueueOptions = {
 	defaultJobOptions: {
-		removeOnComplete: true
+		attempts: 1,
+		removeOnComplete: true,
+		removeOnFail: true
```

Comment:
whoa. That means redis has been keeping all the failed jobs in memory forever?

Reply:
I'm not certain because there was some places referencing `removeOnFail`.  The config stuff is everywhere and super confusing.  I've tried to make it a bit more succinct by having the config in only one place.